docs/guides/unassigning_policies.md

```diff
@@ -0,0 +1,65 @@
+---
+page_title: "Unassigning policies"
+subcategory: ""
+description: |-
+
+---
+# Unassigning policies
+
+For some objects, like network policies, Snowflake [docs](https://docs.snowflake.com/en/sql-reference/sql/drop-network-policy#usage-notes) suggest that a network policy cannot be dropped successfully if it is currently assigned to another object. Currently, the provider does not unassign such objects automatically.
```

Comment:
Added a similar note to network rule, because a rule can't be deleted with `DROP NETWORK RULE` when it's assigned to a network policy.

The behavior you described is interesting, as it's not happening with other policies. Added more notes to databases and schemas.